hazelcast/serialization/portable/reader.py

```diff
@@ -218,12 +239,62 @@ def read_portable_array(self, field_name):
         finally:
             self._in.set_position(current_pos)
 
+    def read_decimal_array(self, field_name):
+        return self.read_object_array_field(
+            field_name, FieldType.DECIMAL_ARRAY, IOUtil.read_big_decimal
+        )
+
+    def read_time_array(self, field_name):
+        return self.read_object_array_field(field_name, FieldType.TIME_ARRAY, read_portable_time)
+
+    def read_date_array(self, field_name):
+        return self.read_object_array_field(field_name, FieldType.DATE_ARRAY, read_portable_date)
+
+    def read_timestamp_array(self, field_name):
+        return self.read_object_array_field(
+            field_name, FieldType.TIMESTAMP_ARRAY, read_portable_timestamp
+        )
+
+    def read_timestamp_with_timezone_array(self, field_name):
+        return self.read_object_array_field(
+            field_name,
+            FieldType.TIMESTAMP_WITH_TIMEZONE_ARRAY,
+            read_portable_timestamp_with_timezone,
+        )
+
     def read_utf(self, field_name):
         return self.read_string(field_name)
 
     def read_utf_array(self, field_name):
         return self.read_string_array(field_name)
 
+    def read_nullable_field(self, field_name, field_type, func):
+        current_pos = self._in.position()
+        try:
+            pos = self._read_position(field_name, field_type)
+            self._in.set_position(pos)
+            is_null = self._in.read_boolean()
+            if is_null:
+                return None
+            return func(self._in)
+        finally:
+            self._in.set_position(current_pos)
+
+    def read_object_array_field(self, field_name, field_type: FieldType, func):
+        current_pos = self._in.position()
+        try:
+            pos = self._read_position(field_name, field_type)
+            self._in.set_position(pos)
+            length = self._in.read_int()
+            if length == NULL_ARRAY_LENGTH:
+                return None
+            values = []
+            for i in range(length):
+                values.append(func(self._in))
```

Comment:
Java does 
```
                    int pos = in.readInt(offset + i * Bits.INT_SIZE_IN_BYTES);
                    in.position(pos);
```
before reading the fields. Don't we need that as well?

Reply:
The same issue with `write_object_array_field` which is explained here https://github.com/hazelcast/hazelcast-python-client/pull/569#discussion_r977557414.

Ported Node.js code is here [DefaultPortableReader.ts#L270](https://github.com/hazelcast/hazelcast-nodejs-client/blob/master/src/serialization/portable/DefaultPortableReader.ts#L270).